### PR TITLE
Add Start-EditorServices script from vscode-powershell repo

### PR DESCRIPTION
**Pull request by rjmholt**
Adds the Start-EditorServices.ps1 script into the actual PowerShellEditorServices repo, to make PSES self-contained

### CODE REVIEW

**Comment by tylerl0706** [comment]
Sorry this can actually replace the contents of this:
https://github.com/PowerShell/PowerShellEditorServices/blob/master/module/Start-EditorServices.ps1

**Comment by rkeithhill** [comment]
Just make sure we keep the version with the recently added logging info.  :-)

**Comment by tylerl0706** [comment]
@rkeithhill yeah that PR was in vscode-powershell :)

**Comment by tylerl0706** [comment]
There's 1 test failure I'm trying to figure out. Not sure if it's a bad test or not.

**Comment by tylerl0706** [comment]
wait... that's not running all the tests.

**Comment by tylerl0706** [comment]
I got them all to run with this:
https://ci.appveyor.com/project/PowerShell/powershelleditorservices/build/1.6.0.1110/tests but there are a lot of failures... but they're related

File: test/PowerShellEditorServices.Test.Host/ServerTestsBase.cs [hunk 1]
```diff
@@ -11,6 +11,7 @@
 using System.Diagnostics;
 using System.IO;
 using System.Text;
```
Comment (rkeithhill) [inline test/PowerShellEditorServices.Test.Host/ServerTestsBase.cs]: Can you add this right here:
```
#if CoreCLR
using System.Reflection;
#endif
```
Without this, my build within VS 2017 fails with:
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS1061	'Type' does not contain a definition for 'GetTypeInfo' and no extension method 'GetTypeInfo' accepting a first argument of type 'Type' could be found (are you missing a using directive or an assembly reference?)	PowerShellEditorServices.Test.Host(netcoreapp2.0)	C:\Users\Keith\GitHub\rkeithhill\PowerShellEditorServices\test\PowerShellEditorServices.Test.Host\ServerTestsBase.cs	42	Active
```

**Comment by tylerl0706** [comment]
@rkeithhill IS THE BEST

**Comment by rjmholt** [comment]
@rkeithhill Wait, so do we want `System.Reflection` but don't want to wrap it with `#if CoreCLR`?

**Comment by rjmholt** [comment]
Oh, I see, @tylerl0706 already fixed it. Just got rid of that last commit

**Comment by rjmholt** [comment]
LGTM

**Comment by tylerl0706** [comment]
Alright, I'm going to merge this in since we currently use the Start-EditorServices.ps1 file so we know that works and the tests are all passing.

**Pull request merged**